src/main/java/Calculations/GPACalculation.java

```diff
@@ -0,0 +1,167 @@
+package Calculations;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.Math;
+
+/**
+ * Construct GPACalculations that consist of different methods for various functionality ranging from computation
+ * of percentage to GPA, converting list marks to list of GPA marks, computing the average of marks,
+ * computation of percentage to lettered Grade, converting list marks to list of lettered Grades, and overall GPA
+ * consisting of GPA and weight. GPA scale and lettered Grade are representive of UofT grading scheme.
+ */
+
+public class GPACalculation {
+
+    /**
+     * Takes a double of mark in terms of percentage and converts to double in terms of GPA point scale
+     * @param percentage is a double that represents the mark in a course
+     * @return a double of GPA value
+     */
+
+    public static double computationGPA(double percentage){
+        if(percentage >= 85.0){
+            return 4.0;
+        }else if(percentage >= 80.0){
+            return 3.7;
+        }else if(percentage >= 77.0){
+            return 3.3;
+        }else if(percentage >= 73.0){
+            return 3.0;
+        }else if(percentage >= 70.0){
+            return 2.7;
+        }else if(percentage >= 67.0){
+            return 2.3;
+        }else if(percentage >= 63.0){
+            return 2.0;
+        }else if (percentage >=60.0){
+            return 1.7;
+        }else if (percentage >= 57.0){
+            return 1.3;
+        }else if (percentage >= 53.0){
+            return 1.0;
+        }else if (percentage >= 50.0){
+            return 0.7;
+        }else {
+            return 0.0;
+        }
```

Comment:
Not a pressing issue, but consider refactoring this function to use some kind of abstract gpa evaluation system. As it is written, this function would be hard to extend to other schools and institutions and violates the open/closed principle.